### PR TITLE
feat(module:tabs): support destroy inactive lazy loaded tab when tabs change

### LINKED ISSUE #8591: Destroy inactive lazy loaded nz-tab
## What problem does this feature solve?
The behavior of nz-tab has changed in [sync ant-design v4.23.6 PR](https://github.com/NG-ZORRO/ng-zorro-antd/pull/8482/files#diff-beb23e4688da2850dbffb292a84b509eaa9aafc7c25c8eab18e1405306f4c97aR26)

Previously, when a lazy-loaded tab is inactive, then its content is destroyed. This has been changed as shown in the link above to keep the content rendered even if the tab is inactive.

This has been a breaking change for some components in our app that were relying on ngOnInit logic executed when the tab is activated for example.

But more importantly, we have some tabs that have active websocket subscriptions that can be resource intensive and ideally they shouldn't be polling data or keeping that subscription to the backend active when their content is not visible.

Can you please provide a suitable replacement for the old behavior.

## What does the proposed API look like?
```html
<nz-tabset>
  <nz-tab nzTitle="Lazy loaded persistent tab">
    <ng-template nz-tab>
      <app-child-component></app-child-component>
    </ng-template>
  </nz-tab>
  <nz-tab nzTitle="Lazy loaded ephemeral tab">
    <ng-template nz-tab let-active>
    @if (active) {
      <app-data-polling-component></app-data-polling-component>
    }
    </ng-template>
  </nz-tab>
</nz-tabset>
```

OR 

```html
<nz-tabset>
  <nz-tab nzTitle="Lazy loaded persistent tab">
    <ng-template nz-tab>
      <app-child-component></app-child-component>
    </ng-template>
  </nz-tab>
  <nz-tab nzTitle="Lazy loaded ephemeral tab">
    <ng-template nz-tab [nzDestroyOnClose]="true">
      <app-data-polling-component></app-data-polling-component>
    </ng-template>
  </nz-tab>
</nz-tabset>
```

### PR DESCRIPTION
**Pull request by Zz-ZzzZ**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: Closes #8591

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: components/tabs/tab-body.component.ts [hunk 1]
```diff
@@ -46,16 +46,29 @@ export class NzTabBodyComponent implements OnChanges {
...
+  /**
+   * When the destroyInactive property is true, this tab will be destroyed when active property change
+   */
+  protected hasBeenDestroy = false;
 
   ngOnChanges(changes: SimpleChanges): void {
     const { active } = changes;
-    if (active?.currentValue) {
-      this.hasBeenActive = true;
+    if (typeof active.currentValue === 'boolean') {
```
Comment (HyperLife1119) [inline components/tabs/tab-body.component.ts:62]:
```suggestion
    if (typeof active?.currentValue === 'boolean') {
```

**Comment by HyperLife1119** [comment]
Can you update this demo?
https://ng.ant.design/components/tabs/zh#components-tabs-demo-lazy

**Comment by Zz-ZzzZ** [comment]
> Can you update this demo? https://ng.ant.design/components/tabs/zh#components-tabs-demo-lazy

sure, do I need to add demo for nzDestroyInactiveTabPane?

**Comment by HyperLife1119** [comment]
You need to change this demo, the demo now needs to be combined with nzDestroyInactiveTabPane to work properly.

**Comment by Zz-ZzzZ** [comment]
Sorry, there are some problems with the previous approach, which is my misunderstanding. the api added by this PR is supposed to destroy inactive lazy loaded tabs. 

Now I have changed the api name and added the demo usage instructions, please check

**Comment by HyperLife1119** [comment]
We need to keep the naming the same as antd, including the behavior, so you don't have to change the naming.

**Comment by Zz-ZzzZ** [comment]
If the behavior needs to be fully synchronized with Antd, then I need to adjust the lazy-loaded policy, because in antd, destroyInactiveTabPane applies to all tab, so I should turn off this PR, which currently applies only to lazy-loaded tab

**Comment by HyperLife1119** [comment]
I'm closing this as #8845 provides a more accurate implementation :)

**Pull request closed without merging**